### PR TITLE
implement SharedPref

### PR DESCRIPTION
**Pull request by MaximeZmt**
# UserSharedPref
### This is related to issue https://github.com/MaximeZmt/SDP_2022-Vibester/issues/142
This Pull Requests contains:

- model.UserSharedPref: Static class that managed the shared preferences for the whole app, using User abstraction
- activity.WelcomeActivity: Small update with the text indicating if logged in or not
- database.UserRepo: Adding a method to update int field in db.
</img>

In addition the code is tested with a coverage of: [96]%

### Notes for reviewers
- Big update of ProfileActivityTest
- Currently at the end of game, update only +1 on number of game but other stat will be done in future
- Still hardcoded a lot, has to be connected to Account creation

### CODE REVIEW

File: app/src/main/java/ch/sdp/vibester/database/UsersRepo.kt [hunk 1]
```diff
@@ -19,16 +19,30 @@ class UsersRepo @Inject constructor() {
...
             .setValue(newVal)
     }
 
     /**
+     * The users class which handled all the interactions with the database that are linked to users
+     * @param userID the id of the user which is being updated
+     * @param newVal (Int) the new value of the field that is being updated
+     * @param fieldName the field name of the field that is being updated
+     */
+    fun updateFieldInt(userID: String, newVal: Int, fieldName: String) {
```
Comment (laurislopata) [inline app/src/main/java/ch/sdp/vibester/database/UsersRepo.kt:37]: Maybe we can pass newVal as a generic type and have one function? What do you think?

**Review by laurislopata — APPROVED** [review]
Just couple of very small things but overall good work! LGTM

File: app/src/main/java/ch/sdp/vibester/database/UsersRepo.kt (hunk 1 above)
Comment (laurislopata) [reply app/src/main/java/ch/sdp/vibester/database/UsersRepo.kt:37]: Like in this example
image

**Review by zwierski — APPROVED** [review]
Good job! Everything looks good to me.

File: app/src/main/java/ch/sdp/vibester/database/UsersRepo.kt (hunk 1 above)
Comment (MaximeZmt) [reply app/src/main/java/ch/sdp/vibester/database/UsersRepo.kt:37]: I think that might be a good idea for a future refactor. Like this is not important yet would be done to improve maintainability.

**Pull request merged**